File: ai/nvidia_deepseek_client.py
import os
from openai import OpenAI

client = OpenAI(
    base_url="https://integrate.api.nvidia.com/v1",
    api_key=os.environ.get("NVIDIA_API_KEY")
)

def send_deepseek(prompt: str) -> str:
    completion = client.chat.completions.create(
        model="deepseek-ai/deepseek-v4-pro",
        messages=[{"role": "user", "content": prompt}],
        temperature=1,
        top_p=0.95,
        max_tokens=16384,
        extra_body={"chat_template_kwargs": {"thinking": False}},
        stream=False
    )
    return completion.choices[0].message.content

File: artifacts/api-server/src/lib/ai/nvidiaDeepseekClient.ts
import OpenAI from 'openai';

export function getDeepseekClient(): OpenAI {
    const apiKey = process.env.NVIDIA_API_KEY;
    if (!apiKey) {
        throw new Error("NVIDIA_API_KEY environment variable is not set");
    }
    return new OpenAI({
        baseURL: "https://integrate.api.nvidia.com/v1",
        apiKey: apiKey,
    });
}

export async function sendDeepseek(prompt: string): Promise<string> {
    const client = getDeepseekClient();
    const completion = await client.chat.completions.create({
        model: "deepseek-ai/deepseek-v4-pro",
        messages: [{ role: "user", content: prompt }],
        temperature: 1,
        top_p: 0.95,
        max_tokens: 16384,
        // @ts-ignore - extra_body is supported by NVIDIA's OpenAI-compatible endpoint
        extra_body: { chat_template_kwargs: { thinking: false } },
        stream: false,
    });
    return completion.choices[0].message.content || "";
}
